src/lxc/conf.c

```diff
@@ -2217,6 +2217,12 @@ static int parse_mntopt(char *opt, unsigned long *flags, char **data, size_t siz
 {
 	size_t ret;
 
+	/* Propagation opts are parsed separately and must not go into data. */
```

Comment:
Good catch!

Just one small question here:
Have you tried to reproduce this issue on a recent LXC versions?
I have tried to add `lxc.mount.entry = tmpfs foobar tmpfs rw,rprivate,noexec,nosuid,nodev,create=dir` to reproduce this and haven't succeed. And as I can see from the code this function is called only from a few codepath like storage drivers (btrfs,zfs,overlayfs).